src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -1145,6 +1151,17 @@ private void getFieldInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_MET
                 ThrowHelper.ThrowInvalidProgramException(ExceptionStringID.InvalidProgramCallVirtStatic, originalMethod);
             }
 
+            if ((flags & CORINFO_CALLINFO_FLAGS.CORINFO_CALLINFO_LDFTN) != 0
+                && originalMethod.IsNativeCallable)
+            {
+                if (!originalMethod.Signature.IsStatic // Must be a static method
+                    || originalMethod.HasInstantiation || originalMethod.OwningType.HasInstantiation // No generics involved
+                    || Marshaller.IsMarshallingRequired(originalMethod)) // Only blittable arguments
```

Comment:
I think this needs to be the `Marshaller.IsMarshallingRequired(originalMethod.Signature)` overload. The one that takes `MethodDesc` expects the MethodDesc to be a p/invoke and asserts that as the first thing.